src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/unsubscribe_button_response_test.cc

```diff
@@ -0,0 +1,148 @@
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <string>
+
+#include "application_manager/mock_event_dispatcher.h"
+#include "gtest/gtest.h"
+#include "hmi/unsubscribe_button_response.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace hmi_commands_test {
+namespace unsubscribe_button_response {
+
+namespace am = ::application_manager;
+using application_manager::ExpiredButtonRequestData;
+using application_manager::commands::ResponseFromHMI;
+using sdl_rpc_plugin::commands::hmi::UnsubscribeButtonResponse;
+using ::test::components::event_engine_test::MockEventDispatcher;
+typedef std::shared_ptr<ResponseFromHMI> ResponseFromHMIPtr;
+
+namespace {
+const uint32_t kCorrelationId = 2u;
+const uint32_t kAppId = 1u;
+const hmi_apis::Common_ButtonName::eType kButtonName =
+    hmi_apis::Common_ButtonName::SEEKLEFT;
+const hmi_apis::FunctionID::eType kFunctionID =
+    hmi_apis::FunctionID::Buttons_UnsubscribeButton;
+}  // namespace
+
+class HMIUnsubscribeButtonResponseTest
+    : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  HMIUnsubscribeButtonResponseTest() {
+    ON_CALL(app_mngr_, event_dispatcher())
+        .WillByDefault(ReturnRef(event_dispatcher_));
+  }
+
+  MessageSharedPtr CreateCommandMsg() {
+    MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+    (*command_msg)[strings::params][strings::correlation_id] = kCorrelationId;
+    (*command_msg)[am::strings::params][am::hmi_response::code] =
+        hmi_apis::Common_Result::SUCCESS;
+
+    return command_msg;
+  }
+
+  MockEventDispatcher event_dispatcher_;
+};
+
+TEST_F(HMIUnsubscribeButtonResponseTest,
+       RUN_ProcessExpiredRequest_SendUnsubscribeRequestToHMI) {
```

Comment:
And here also SendSubscribeRequestToHMI and check all arguments in CreateButtonSubscriptionHandlingRequestToHmi(_, _, _, _)) to prevent test malfunction